modules/apps/transfer/types/denom.go

```diff
@@ -11,6 +11,14 @@ import (
 	cmtbytes "github.com/cometbft/cometbft/libs/bytes"
 )
 
+// NewDenom creates a new Denom instance given the base denomination and trace.
+func NewDenom(base string, trace ...Trace) Denom {
```

Comment:
I thought it would be nice to have a constructor function. I can use it other places of the codebase, once the PR is merged.